src/main/java/org/sagebionetworks/bridge/notification/helper/TaskHistoryIterator.java

```diff
@@ -0,0 +1,100 @@
+package org.sagebionetworks.bridge.notification.helper;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import org.joda.time.DateTime;
+
+import org.sagebionetworks.bridge.rest.ClientManager;
+import org.sagebionetworks.bridge.rest.api.ForWorkersApi;
+import org.sagebionetworks.bridge.rest.model.ForwardCursorScheduledActivityList;
+import org.sagebionetworks.bridge.rest.model.ScheduledActivity;
+
+/** Helper class that abstracts away Bridge's paginated API and uses an iterator instead. */
+public class TaskHistoryIterator implements Iterator<ScheduledActivity> {
+    // Package-scoped for unit tests
+    static final int PAGE_SIZE = 10;
+
+    // Instance invariants
+    private final ClientManager clientManager;
+    private final String studyId;
+    private final String userId;
+    private final String taskId;
+    private final DateTime scheduledOnStart;
+    private final DateTime scheduledOnEnd;
+
+    // Instance state tracking
+    private ForwardCursorScheduledActivityList taskList;
+    private int nextIndex;
+
+    /**
+     * Constructs a TaskHistoryIterator for the given Bridge client, study, user, task ID, and schedule bounds. This
+     * kicks off requests to load the first page.
+     */
+    public TaskHistoryIterator(ClientManager clientManager, String studyId, String userId, String taskId,
```

Comment:
Does this only load tasks, and not surveys or compound tasks. Do we know this is the only thing they'll want?

Reply:
This is consistent with the burst schedule currently configured in mPower 2. We have a single task ID that encompasses all 4 tasks (tapping, walking, tremor, and ???).

I don't think we have any value in having granular notifications over a set of tasks. (At least not for mPower 2. The more complicated we make this, the less configurable it'll be.) Using Survey GUIDs would be kind of a pain, and afaict Compound Tasks are a dead feature.

It might be worth discussing this in a further design meeting, but at best, that would be a separate JIRA that we would prioritize accordingly against other tasks.